### PR TITLE
#602: Store scheduler config in DB to prevent auto schedule on every …

### PR DESCRIPTION
**Pull request by kkaczmarczyk**
…deploy

### CODE REVIEW

File: src/main/java/org/benetech/servicenet/config/QuartzConfig.java [hunk 1]
```diff
@@ -47,35 +43,22 @@
...
     public SchedulerFactoryBean schedulerFactoryBean() {
-        SchedulerFactoryBean scheduler = new SchedulerFactoryBean();
+        SchedulerFactoryBean schedulerFactoryBean = new SchedulerFactoryBean();
 
         AutowiringBeanJobFactory jobFactory = new AutowiringBeanJobFactory();
         jobFactory.setApplicationContext(applicationContext);
-        scheduler.setJobFactory(jobFactory);
-
-        List<BaseJob> jobs = new ArrayList<>();
-        jobs.add(edenDataUpdateJob);
```
Comment (pmuchowski) [inline src/main/java/org/benetech/servicenet/config/QuartzConfig.java]: you can remove the autowired jobs if they are not used anymore.

File: src/main/java/org/benetech/servicenet/service/impl/SchedulerServiceImpl.java [hunk 2]
```diff
@@ -52,6 +54,31 @@
...
+        jobs.add(uwbaTaxonomyUpdateJob);
+
+        jobs
+            .forEach(job -> {
+                try {
+                    if (scheduler.getTrigger(job.getInitTrigger().getKey()) == null) {
+                        scheduler.scheduleJob(edenDataUpdateJob.getJobDetail(), edenDataUpdateJob.getInitTrigger());
+                    }
+                } catch (SchedulerException e) {
+                    log.error(e.getMessage());
```
Comment (pmuchowski) [inline src/main/java/org/benetech/servicenet/service/impl/SchedulerServiceImpl.java]: I think you should log the entire exception `log.error(e.getMessage(), e);`

File: src/main/resources/config/liquibase/tables_postgres.sql [hunk 3]
```diff
@@ -0,0 +1,187 @@
+-- Thanks to Patrick Lightbody for submitting this...
+--
+-- In your Quartz properties file, you'll need to set 
+-- org.quartz.jobStore.driverDelegateClass = org.quartz.impl.jdbcjobstore.PostgreSQLDelegate
```
Comment (pmuchowski) [inline src/main/resources/config/liquibase/tables_postgres.sql]: where is this file used?

File: src/main/java/org/benetech/servicenet/config/QuartzConfig.java (hunk 1 above)
Comment (kkaczmarczyk) [reply src/main/java/org/benetech/servicenet/config/QuartzConfig.java]: fixed

File: src/main/java/org/benetech/servicenet/service/impl/SchedulerServiceImpl.java (hunk 2 above)
Comment (kkaczmarczyk) [reply src/main/java/org/benetech/servicenet/service/impl/SchedulerServiceImpl.java]: fixed

File: src/main/resources/config/liquibase/tables_postgres.sql (hunk 3 above)
Comment (kkaczmarczyk) [reply src/main/resources/config/liquibase/tables_postgres.sql]: it's meant for a re-creation of Quartz DB tables - I have added info to the README.

**Pull request merged**